### PR TITLE
Add Prometheus data source guide

### PR DESCRIPTION
**Pull request by hemajv**
## Related Issues and Dependencies
#11 

## This introduces a breaking change

- [ ] Yes
- [X] No

## Description

Adding a prometheus data source guide

### CODE REVIEW

**Review by oindrillac — COMMENTED** [review]
/lgtm 👍

File: data-sources/prometheus.md [hunk 1]
```diff
@@ -1,12 +1,78 @@
...
+
+Shown above is the Prometheus UI which allows you to search and aggregate different time series metrics and view them as a graph.
+
+## Purpose and Use of Prometheus
+
+Prometheus monitoring has become the industry standard and is implemented by software companies worldwide. It supports multidimensional data collection and data queuing, and is reliable, allowing customers to quickly diagnose problems.
+
+**Operations Perspective** - From an operations standpoint, Prometheus is useful in tracking the performance of various services/applications and enables SRE teams to effectively diagnose, troubleshoot and handle incidents for any degradation of services.
+
+**Data Science Perspective** - From a Data Science standpoint, Prometheus provides a large collection of time series data which can be made available to Data Scientists for exploring and applying ML methods to solve unique problems such as time series forecasting to predict when a possible system failure may occur in the future or anomaly detection i.e identifying any erratic or erroneous system behavior.
```
Comment (Shreyanand) [inline data-sources/prometheus.md:23]: From a Data Science standpoint, Prometheus provides a large collection of time series data for exploring and applying ML methods to solve unique problems such as forecasting to predict a future system failure or anomaly detection to identify erratic or erroneous system behavior.

File: data-sources/prometheus.md [hunk 2]
```diff
@@ -1,12 +1,78 @@
...
+* [API Client](https://github.com/AICoE/prometheus-api-client-python)
+
+* [Example Jupyter Notebook](https://github.com/AICoE/prometheus-api-client-python/blob/master/examples/MetricsList_example.ipynb) - This example Jupyter notebook explains the API client library and its functions supported
+
+* [Tutorial](https://learn.openshift.com/ai-machine-learning/prometheus-api-client/) - Interactive tutorial on using the Prometheus API client library
 
 ## Caveats
 
-* beware combanatory explosion of labels
+* As Prometheus metrics have many labels, be careful when combining these labels with other metrics as they can result in an explosion of metric label pairs, making it complex and compute intensive to query
```
Comment (Shreyanand) [inline data-sources/prometheus.md:72]: As Prometheus metrics have many labels, they have to be carefully combined with other metrics...

**Review by Shreyanand — CHANGES_REQUESTED** [review]
Looks good, minor edits :+1:

File: data-sources/prometheus.md [hunk 3]
```diff
@@ -1,12 +1,78 @@
...
+
+**Example Prometheus Instance**: http://demo.robustperception.io:9090/graph
+
+![prometheus-timeseries-graph](../assets/images/prometheus-timeseries-graph.png)
+
+Shown above is the Prometheus UI which allows you to search and aggregate different time series metrics and view them as a graph.
+
+## Purpose and Use of Prometheus
+
+Prometheus monitoring has become the industry standard and is implemented by software companies worldwide. It supports multidimensional data collection and data queuing, and is reliable, allowing customers to quickly diagnose problems.
```
Comment (chauhankaranraj) [inline data-sources/prometheus.md:19]:
```suggestion
Prometheus has become the industry standard for monitoring, and is implemented by software companies worldwide. It supports multidimensional data collection and data queuing, and is reliable, allowing customers to quickly diagnose problems.
```

**Review by chauhankaranraj — COMMENTED** [review]
Tiny nit but otherwise looks great! :)

File: data-sources/prometheus.md (hunk 3 above)
Comment (hemajv) [reply data-sources/prometheus.md]: updated :+1:

File: data-sources/prometheus.md (hunk 2 above)
Comment (hemajv) [reply data-sources/prometheus.md]: updated

**Comment by hemajv** [comment]
Thanks for the reviews folks, updated the changes! cc @Shreyanand @chauhankaranraj

**Review by chauhankaranraj — COMMENTED** [review]
/lgtm

**Review by Shreyanand — APPROVED** [review]
👍  /lgtm

**Review by aakankshaduggal — COMMENTED** [review]
/lgtm

**Comment by MichaelClifford** [comment]
/lgtm 
:rocket:

**Pull request merged**